File: Java/RecursiveCall/BJAlgo_2447.java
import java.io.*;
import java.util.*;

public class BJAlgo_2447 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        List stars = new ArrayList();
        stars.add("***");
        stars.add("* *");
        stars.add("***");

        int k = 0;
        while (N != 3) {
            N = N / 3;
            k++;
        }

        for (int i = 0; i < k; i++) {
            stars = star(stars);
        }

        for (int i = 0; i < stars.size(); i++) {
            System.out.println(stars.get(i));
        }

    }

    private static List star(List list) {
        List tmp = new ArrayList();
        for (int i = 0; i < 3 * list.size(); i++) {
            StringBuilder sb = new StringBuilder();
            if (i / list.size() == 1) {
                for (int j = 0; j < list.size(); j++) {
                    sb.append(" ");
                }
                tmp.add(list.get(i % list.size()) + "" + sb + list.get(i % list.size()));
            } else {
                for (int j = 0; j < 3; j++) {
                    sb.append(list.get(i % list.size()));
                }
                tmp.add(sb);
            }
        }

        return tmp;
    }
}

File: Python/RecursiveCall/BJAlgo_2447.py
from sys import stdin


def star(num):
    matrix = []
    for i in range(3 * len(num)):
        # 3으로 나눈 몫이 1인 경우에만 중앙이 뚫린 형태
        if i // len(num) == 1:
            matrix.append(num[i % len(num)] + (" " * len(num)) + num[i % len(num)])
        # 나머지 경우는 처음 구조를 그대로 반복
        else:
            matrix.append(num[i % len(num)] * 3)

    return list(matrix)


N = int(stdin.readline())
stars = ["***", "* *", "***"]
k = 0

while N != 3:
    N = int(N/3)
    k += 1

for i in range(k):
    stars = star(stars)

for i in range(len(stars)):
    print(stars[i])